tests/tests/htlc_tests.cpp

```diff
@@ -177,6 +177,448 @@ try {
+      trx.clear();
+   }   
+   // Alice attempts to put a contract on the blockchain that is correct but has a preimage size of 0
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+      create_operation.to = bob_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::sha256>( pre_image );
+      create_operation.preimage_size = 0;
+      create_operation.from = alice_id;
+      create_operation.extensions.value.memo = memo_data();
+      create_operation.extensions.value.memo->from = alice_public_key;
+      create_operation.extensions.value.memo->to = bob_public_key;
+      create_operation.extensions.value.memo->set_message(
+            alice_private_key, bob_public_key, "Dear Bob,\n\nMoney!\n\nLove, Alice");
+      create_operation.fee = db.get_global_properties().parameters.current_fees->calculate_fee(create_operation);
+      trx.operations.push_back(create_operation);
+      sign(trx, alice_private_key);
+      GRAPHENE_CHECK_THROW(PUSH_TX(db, trx, ~0), fc::exception);
+      trx.clear();
+   }
+   // Alice attempts to put a contract on the blockchain with HASH160
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+      create_operation.to = bob_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::hash160>( pre_image );
+      create_operation.preimage_size = preimage_size;
+      create_operation.from = alice_id;
+      create_operation.fee = db.get_global_properties().parameters.current_fees->calculate_fee(create_operation);
+      trx.operations.push_back(create_operation);
+      sign(trx, alice_private_key);
+      // this try/catch is just to get some console output to verify the exception
+      try {
+         PUSH_TX(db, trx, ~0);
+         BOOST_FAIL("crate_operation should have failed due to the HASH160 hash");
+      } catch(fc::exception& ex) {
+         if ( ex.to_string().find("HASH160 unavailable") == std::string::npos )
+            BOOST_FAIL("create_operation failed but not due to the hash160 function");
+       }
+      trx.clear();
+   }
+   // Alice attempts to put a contract on the blockchain with a memo
+   {
+      graphene::chain::htlc_create_operation create_operation;
+      BOOST_TEST_MESSAGE("Alice (who has 100 coins, is transferring 3 coins to Bob");
+      create_operation.amount = graphene::chain::asset( 3 * GRAPHENE_BLOCKCHAIN_PRECISION );
+      create_operation.to = bob_id;
+      create_operation.claim_period_seconds = 60;
+      create_operation.preimage_hash = hash_it<fc::hash160>( pre_image );
```

Comment:
Should not use `hash160` here, since in the previous test case we've made sure it will fail anyway regardless of whether a memo is there.